code/modules/games/cards.dm

```diff
@@ -95,8 +107,8 @@
 		to_chat(user,"<span class='notice'>There are no cards in the deck.</span>")
 		return
 
-	var/obj/item/weapon/hand/H = user.get_type_in_hands(/obj/item/weapon/hand)
-	if(H && !(H.parentdeck == src))
+	var/obj/item/weapon/hand/H = user.get_active_hand(/obj/item/weapon/hand)
```

Comment:
Noting here our conversations about how a `hand` is a special case of a `deck`. What I really dislike about this, though, is that `/obj/item/weapon/hand` tells me exactly nothing about what it is. Put `card` somewhere in that typepath so it's apparent.
Also, don't subtype these off of weapon, I'm trying to kill that type.